src/lib/config/ModelConfig.cc

```diff
@@ -424,13 +424,19 @@ void ModelConfig::setExpectations(bool isDefault) {
   expectations_.addChild(
       ExpectationNode::createExpectation("Branch-Predictor"));
 
+  expectations_["Branch-Predictor"].addChild(
+      ExpectationNode::createExpectation<std::string>(
+          "Generic", "Type"));
+  expectations_["Branch-Predictor"]["Type"].setValueSet(
+      std::vector<std::string>{"Generic", "Perceptron"});
+
   expectations_["Branch-Predictor"].addChild(
       ExpectationNode::createExpectation<uint8_t>(8, "BTB-Tag-Bits"));
   expectations_["Branch-Predictor"]["BTB-Tag-Bits"].setValueBounds<uint8_t>(1,
                                                                             64);
-
   expectations_["Branch-Predictor"].addChild(
-      ExpectationNode::createExpectation<uint8_t>(2, "Saturating-Count-Bits"));
+      ExpectationNode::createExpectation<uint8_t>(2, "Saturating-Count-Bits",
+                                                  true));
   expectations_["Branch-Predictor"]["Saturating-Count-Bits"]
       .setValueBounds<uint8_t>(1, 64);
```

Comment:
I'm not sure how I feel about the first comment - given all config options have a default value now it doesn't matter if extra are set / if they're not set. Although it could be useful to determine exactly what should and shouldn't be set.... (thinking out loud on this one)

I disagree with the second point - this is what documentation is for and if they are redunant then they won't be used in simulation anyway. Perhaps we need to update the current documentation to make it clearer regarding what an emulation core does / will use from the config

Reply:
Happy with either option here.  Subject to further discussion, I've added to the documentation for the BP config options that Saturating-Count-Bits and Fallback-Static-Predictor are only needed for the GenericPredictor.

Reply:
@dANW34V3R The first point could go either way. There's no programming or performance advantage in having it be more conditional but it could help with code clarity and is technically more "correct". The second point, outputting the redundant variables on each run wouldn't be a viable solution as it's just too messy in the STDOUT. A solution similar to Finn's comment of moving it over to the docs would be a better solution currently.

@JosephMoore25 I like the idea of a matrix. It would have to be a 4-column, n-row (with n being the number of config options) matrix to be consistently readable. The inclusion of ranges in the docs would be advantageous. Also, the matrix idea helps with giving the user knowledge of what is needed before running SimEng (similar to your comment on the limits)

Reply:
My main (first) point here was about correctness and clarity rather than anything else. It would function yes, but a condition would make it clear to a dev.

Ranges/ valid options should be a given in the docs I think. I can create an issue if we all agree so that we don't forget to do it. 

Another idea to help this could be to add a skeleton config with all possible options and ranges either to the docs or the configs folder. Potentially this can be generated from the expectations so no upkeep needed and should make it very easy for a new user to create a new config and get started straight away.

A matrix could be nice, potentially might be quite large but would need to just try it and see. We can discuss the exact format in the meeting tomorrow.

Reply:
Offline discussion lead to decision to include `if`s in model config. I will resolve this comment once done

No offline discussion but it seems there is agreement to add ranges and potentially a matrix to the docs. This shouldn't be part of this PR. I will create an issue summarising this